crates/trie/src/state.rs

```diff
@@ -38,17 +39,27 @@ impl HashedPostState {
         state: impl IntoIterator<Item = (&'a Address, &'a BundleAccount)>,
     ) -> Self {
         let mut this = Self::default();
-        for (address, account) in state {
-            let hashed_address = keccak256(address);
-            this.accounts.insert(hashed_address, account.info.clone().map(into_reth_acc));
-
-            let hashed_storage = HashedStorage::from_iter(
-                account.status.was_destroyed(),
-                account.storage.iter().map(|(key, value)| {
-                    (keccak256(B256::new(key.to_be_bytes())), value.present_value)
-                }),
-            );
-            this.storages.insert(hashed_address, hashed_storage);
+
+        let changes = state
+            .into_iter()
+            .collect::<Vec<(&'a Address, &'a BundleAccount)>>()
```

Comment:
not possible to avoid this?

Reply:
changed the argument to `&revm_interpreter::primitives::HashMap`, so that we can avoid the `collect()`